src/Quizzer.API.DAL/Repository/Interfaces/IApiRepository.cs

```diff
@@ -0,0 +1,40 @@
+﻿using Quizzer.API.DAL.Entities.Interface;
+using System.Linq.Expressions;
+
+namespace Quizzer.API.DAL.Repository.Interfaces {
+    public interface IRepository<TEntity>
+        where TEntity : class, IEntity {
+        /// <param name="navigationPropertyPath">A string of '.' separated navigation property names to be included.</param>
+        /// <returns>Found entity by it's ID, or null when there is no entity with specified ID</returns>
+        TEntity? GetById(Guid id, string navigationPropertyPath = "");
+
+        /// <param name="navigationPropertyPaths">A collection of strings, which are in form of '.' separated navigation property names to be included.</param>
+        /// <returns>Found entity by it's ID, or null when there is no entity with specified ID</returns>
+        TEntity? GetById(Guid id, IEnumerable<string> navigationPropertyPaths);
+
+        /// <param name="navigationPropertyPath">A string of '.' separated navigation property names to be included.</param>
+        IEnumerable<TEntity> GetByPredicate(Expression<Func<TEntity, bool>> predicate, string navigationPropertyPath = "");
+
+        /// <param name="navigationPropertyPaths">A collection of strings, which are in form of '.' separated navigation property names to be included.</param>
+        IEnumerable<TEntity> GetByPredicate(Expression<Func<TEntity, bool>> predicate, IEnumerable<string> navigationPropertyPaths);
+
+        /// <param name="navigationPropertyPaths">A collection of strings, which are in form of '.' separated navigation property names to be included.</param>
+        IEnumerable<TEntity> GetAll(IEnumerable<string>? navigationPropertyPaths = null);
+
+        IEnumerable<TEntity> GetAll();
+
+        void Delete(Guid entityId);
+
+        bool Exists(Guid entityId);
+
+        /// <returns>New entity reference with inserted entity data</returns>
+        TEntity Insert(TEntity entity);
```

Comment:
Why return TEntity on Insert, but return Guid on Update?

Reply:
I missed that during refactor, fixed